src/modules/ekf2/EKF/mag_fusion.cpp

```diff
@@ -75,7 +75,12 @@ bool Ekf::fuseMag(const Vector3f &mag, estimator_aid_source3d_s &aid_src_mag, bo
 		_fault_status.flags.bad_mag_x = true;
 
 		// we need to re-initialise covariances and abort this fusion step
-		resetMagRelatedCovariances();
+		if (update_all_states) {
+			resetQuatCov();
```

Comment:
Side note: I'm actually wondering what's happening if we reset the quat cov like that in air (tilt cov are also reset)